### PR TITLE
[Workers] Clarify thenable support in AsyncLocalStorage API docs

### PR DESCRIPTION
**Pull request by irvinebroque**
refs https://github.com/cloudflare/workerd/issues/870

### CODE REVIEW

File: content/workers/runtime-apis/nodejs/asynclocalstorage.md [hunk 1]
```diff
@@ -227,3 +227,5 @@ console.log(myResource.doSomething()); // prints 123
 * Workers does not implement the full [`async_hooks`](https://nodejs.org/dist/latest-v18.x/docs/api/async_hooks.html) API upon which Node.js' implementation of `AsyncLocalStorage` is built.
 
 * Workers does not implement the ability to create an `AsyncResource` with an explicitly identified trigger context as allowed by Node.js. This means that a new `AsyncResource` will always be bound to the async context in which it was created.
+
+* Thenables (non-Promise objects that expose a `then()` method) are not fully supported when using AsyncLocalStorage.
```
Comment (irvinebroque) [inline content/workers/runtime-apis/nodejs/asynclocalstorage.md]: @jasnell — is there a workaround we should point people towards using `AsyncResource`? https://github.com/cloudflare/workerd/pull/282#issuecomment-1402921124

File: content/workers/runtime-apis/nodejs/asynclocalstorage.md [hunk 2]
```diff
@@ -227,3 +227,5 @@ console.log(myResource.doSomething()); // prints 123
 * Workers does not implement the full [`async_hooks`](https://nodejs.org/dist/latest-v18.x/docs/api/async_hooks.html) API upon which Node.js' implementation of `AsyncLocalStorage` is built.
```
Comment (irvinebroque) [inline content/workers/runtime-apis/nodejs/asynclocalstorage.md:227]:
```suggestion
* Workers does not implement the full [`async_hooks`](https://nodejs.org/dist/latest-v18.x/docs/api/async_hooks.html) API upon which Node.js' implementation of `AsyncLocalStorage` is built.
```

File: content/workers/runtime-apis/nodejs/asynclocalstorage.md (hunk 1 above)
Comment (deadlypants1973) [inline content/workers/runtime-apis/nodejs/asynclocalstorage.md]:
```suggestion
* Thenables (non-Promise objects that expose a `then()` method) are not fully supported when using `AsyncLocalStorage`.
```

**Review by deadlypants1973 — APPROVED** [review]
PCX

File: content/workers/runtime-apis/nodejs/asynclocalstorage.md [hunk 3]
```diff
@@ -227,3 +227,5 @@ console.log(myResource.doSomething()); // prints 123
...
 
 * Workers does not implement the ability to create an `AsyncResource` with an explicitly identified trigger context as allowed by Node.js. This means that a new `AsyncResource` will always be bound to the async context in which it was created.
+
+* Thenables (non-Promise objects that expose a `then()` method) are not fully supported when using `AsyncLocalStorage`.
```
Comment (irvinebroque) [inline content/workers/runtime-apis/nodejs/asynclocalstorage.md]:
```suggestion
* Thenables (non-Promise objects that expose a `then()` method) are not fully supported when using `AsyncLocalStorage`. When working with thenables, instead use [`AsyncLocalStorage.snapshot()`](https://nodejs.org/api/async_context.html#static-method-asynclocalstoragesnapshot) to capture a snapshot of the current context.
```

**Pull request merged**